iothub/device/tests/Pipeline/RetryDelegatingHandlerTests.cs

```diff
@@ -89,30 +89,33 @@ public async Task RetryDelegatingHandler_DoesNotRetryOnNotSupportedException()
         {
             // arrange
             int callCounter = 0;
-
-            PipelineContext contextMock = Substitute.For<PipelineContext>();
-            contextMock.ConnectionStatusChangeHandler = (connectionStatusInfo) => { };
-            IDelegatingHandler nextHandlerMock = Substitute.For<IDelegatingHandler>();
             var message = new TelemetryMessage(new byte[] { 1, 2, 3 });
-            nextHandlerMock.OpenAsync(Arg.Any<CancellationToken>()).Returns(Task.CompletedTask);
+
```

Comment:
I find more and more tests using NSubstitute fail occasionally. When one fails, I update it to use Moq.

Soon, we should make a pass and proactively update the remaining ones and remove NSubstitute as a dependency.